### PR TITLE
rust: Add mentoring note for parallel letter frequency

### PR DESCRIPTION
**Pull request by ivfranco**
#1317

### CODE REVIEW

**Comment by SleeplessByte** [comment]
cc @TheDarkula 

@iHiD can you add Meade to the `website-copy-rust` team?

**Comment by iHiD** [comment]
Done. @TheDarkula What other teams were you part of before the 2FA thing kicked you out?

**Comment by SleeplessByte** [comment]
cc @exercism/rust

File: tracks/rust/exercises/parallel-letter-frequency/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+// test bench_tiny_sequential  ... bench:         164 ns/iter (+/- 4)
+```
+
+Thanks to the type system of Rust, if the solution compiles & passes all the tests, usually there's no data race in it. Students may however question the necessity of these apparently too restrictive constraints, and it's the mentors' responsibility to explain these concepts throughout:
+
+- Why std::thread::spawn requires all captured variables having 'static lifetime: thread in libstd is unscoped, they can live for arbitrarily long & outlive all lifetimes except 'static. It may be helpful to provide an historical note on scoped thread in pre-1.0 Rust and why it's removed: https://github.com/rust-lang/rust/issues/24292. An introduction to scoped thread in external crates (rayon or crossbeam) may also be helpful. 
+- What is Send and why std::thread::spawn requires all captured variables implement Send: for students got this far into the Rust track it's probably enough to direct them to the documentation of [std::sync::Arc](https://doc.rust-lang.org/std/sync/struct.Arc.html). Nomicon also has [a chapter explaining data race](https://doc.rust-lang.org/nomicon/races.html).
```
Comment (efx) [inline tracks/rust/exercises/parallel-letter-frequency/mentoring.md]:
```suggestion
- What is Send and why std::thread::spawn requires all captured variables implement Send: for students who have gotten this far into the Rust track it's probably enough to direct them to the documentation of [std::sync::Arc](https://doc.rust-lang.org/std/sync/struct.Arc.html). Nomicon also has [a chapter explaining data race](https://doc.rust-lang.org/nomicon/races.html).
```

**Pull request merged**